pkg/regionspan/region_range_lock.go

```diff
@@ -40,32 +40,32 @@ func rangeTsEntryWithKey(key []byte) *rangeTsEntry {
 	}
 }
 
-func (e *rangeTsEntry) Less(than btree.Item) bool {
-	return bytes.Compare(e.startKey, than.(*rangeTsEntry).startKey) < 0
+func rangeTsEntryLess(a, b *rangeTsEntry) bool {
+	return bytes.Compare(a.startKey, b.startKey) < 0
 }
 
 // RangeTsMap represents a map from key range to a timestamp. It supports range set and calculating min value among a
 // a specified range.
 type RangeTsMap struct {
-	m *btree.BTree
+	m *btree.BTreeG[*rangeTsEntry]
```

Comment:
Please investigate whether using [NewFreeListG](https://pkg.go.dev/github.com/google/btree#NewFreeListG) will have any benefit here.

Reply:
Benchmark results show having many btrees sharing one big single freelist does not bring significant performance improvements, especially when the number of btrees can be large. The reasons are 
1) The shared freelist in google/btree is protected by a mutex. Multiple irrelevant btrees waiting for the mutex of freelist does no good to performance. 
2) The size of freelist is hard to decide when we don't know how many btrees share one. Here, the number of RangeTsMap can be very big since there can be a large number of tables . We don't get benefits given the size of freelist is no greater than the number of btrees sharing it.